pkg/controller/operatingsystemconfig/actuator.go

```diff
@@ -9,23 +9,35 @@ import (
 	_ "embed"
 	"fmt"
 	"path/filepath"
+	"strings"
+	"text/template"
 
 	"github.com/gardener/gardener/extensions/pkg/controller/operatingsystemconfig"
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
 	"github.com/go-logr/logr"
 	"k8s.io/utils/ptr"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	configv1alpha1 "github.com/gardener/gardener-extension-os-coreos/pkg/controller/config/v1alpha1"
 )
 
 type actuator struct {
-	client client.Client
+	client          client.Client
+	extensionConfig Config
+}
+
+// Config contains configuration for the extension service.
+type Config struct {
```

Comment:
Whats the reason to embed the `ExtensionConfig` in this struct and not directly using the `ExtensionConfig` in the `actuator` struct?

Reply:
personal preference, easier to extend.